homeassistant/components/media_player/cast.py

```diff
@@ -265,10 +264,15 @@ def async_cast_discovered(discover: ChromecastInfo) -> None:
         info = await hass.async_add_job(_fill_out_missing_chromecast_info,
                                         info)
         if info.friendly_name is None:
-            # HTTP dial failed, so we won't be able to connect.
-            raise PlatformNotReady
+            _LOGGER.debug("Cannot retrieve detail information for chromecast"
+                          " %s, the device may not online", info)
+            remove_handler()
+            return False
```

Comment:
Style-wise I find returning a boolean a *bit* bad. `async_setup_platform`s are not supposed to return booleans. Of course `_async_setup_platform` is only a proxy for `async_setup_platform` but it might be good to keep the syntax the same for style. Besides, the old way could actually be a bit nicer too:

```python
async def _async_setup_platform(hass: HomeAssistantType, config: ConfigType,
                                async_add_entities, discovery_info):
    # ...
    if info.friendly_name is None:
        # HTTP dial failed, so we won't be able to connect.
        remove_handler()
        raise PlatformNotReady
    # ...


async def async_setup_entry(hass, config_entry, async_add_entities):
     # ...
     done, pending = await asyncio.wait([
        _async_setup_platform(hass, cfg, async_add_entities, None)
        for cfg in config], timeout=10)
    # ...
```